priv/apps/itest/test/features/in_flight_exits.feature

```diff
@@ -1,6 +1,17 @@
 Feature: In Flight Exits
   Scenario: Alice starts an In Flight Exit
-    When Alice deposits "1" ETH to the root chain
-    Then Alice should have "1" ETH on the root chain after finality margin
-    When Alice starts an in flight exit
-    Then Alice should have "1" ETH after finality margin
\ No newline at end of file
+    When "Alice" deposits "10" ETH to the root chain
+    When "Bob" deposits "10" ETH to the root chain
+    Then "Alice" should have "10" ETH on the child chain after finality margin
+    Then "Bob" should have "10" ETH on the child chain after finality margin
+    When Alice creates a transaction for "5" ETH
+    Then Bob gets in flight exit data for "5" ETH
+    Then Alice sends a transaction tx1
+    Then Bob sends a transaction spending Alices outputs of tx1
+    When Alice starts an in flight exit of the tx1 transaction
+    Then Alice verifies its in flight exit of tx1 transaction
+    Then Bob piggybacks inputs and outputs from Alice
+    Then Bob starts a competing in flight exit
```

Comment:
so similarly, what I just noticed, this leverages the `Bob gets in flight exit data` step, which is a bit hidden. It is only there, where the "competingness" is decided. And that also relies on the `Alice and Bob` from the previous step, and the fact the Bob shared a UTXO to create his txs, so going the "explicit" and "tx1" path, those previous steps would need to explicitly name the UTXOs somehow.

I hear you on the "juggling" part here. I see the alternative from the opposite end of the spectrum to go like this:

```
Feature: In Flight Exits
  Scenario: Alice starts an In Flight Exit
    When "Alice" deposits "10" ETH to the root chain
    When "Bob" deposits "10" ETH to the root chain
    Then "Alice" should have "10" ETH on the child chain after finality margin
    Then "Bob" should have "10" ETH on the child chain after finality margin
    When Alice and Bob create a transaction for "5" ETH from their most recent deposits
    Then Bob creates in flight exit data for "5" ETH from his most recent deposit
    Then Alice sends the most recently created transaction
    Then Bob sends the most recently created transaction
    When Alice starts an in flight exit from the most recently created transaction
    Then Alice verifies its in flight exit from the most recently created transaction
    Then Bob piggybacks inputs and outputs from Alices most recent ife
    Then Bob starts an in flight exit from his most recently created transaction
    Then Alice fully challenges Bobs most recent invalid in flight exit
    Then Alice processes her own most recent in flight exit
```

this sounds a bit ridiculous, but it would allow to avoid the naming tx1, utxo1 etc. Less expressive, very much gauged towards a specific scenario, but maybe?

Reply:
I actually prefer this than referencing transactions like "tx1". Why? Because I'm scared we will get into a position where the API is not going to be able to do an exact mapping of the description.

So your approach actually seems very reasonable, because it allows the description to be decoupled of the passing state (which hopefully is a good thing!). 

So I'll adapt the test to your steps and let's see how that plays out.
Kudos!

Reply:
heh, we just in parallel got off a 1:1 with @achiurizo, inclining more towards referencing (aka `tx1`). Tough call, I think this will take a fair amount of iterations/experiments to nail down.

So pragmatically speaking, I'll just refrain to opting for the "keeping the old test for time being" part, to ensure behavior doesn't accidentally get un-covered, as we go through experimentation  and iterations in here.

Reply:
@InoMurko @pdobacz syncing up here after having a convo with @InoMurko . I think let's go with the former approach (*not* naming/referencing transactions _yet_) and see if that accomplishes what we need for this first pass.

Later down when we have more usage from ourselves, we can refine the language here to for future use cases.